### PR TITLE
Add pre-commit config, Readme instructions and a workflow check.

### PR DESCRIPTION
**Pull request by MichaelsJP**
Pre-commit will make sure that all commits going into master fit the absolute basic checks like a single linting style, good commit messages, trailing whitespaces etc. It makes sure that the quality is kept and commits don't consist of things like fixing empty lines etc.
The workflow ensures that the checks are at least running once before changes are going into main.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+          - "eslint@7.6.0"
+          - "eslint-config-standard@14.1.1"
+          - "eslint-friendly-formatter@3.0.0"
+          - "eslint-loader@4.0.2"
+          - "eslint-plugin-import@2.22.0"
+          - "eslint-plugin-node@11.1.0"
+          - "eslint-plugin-promise@4.2.1"
+          - "eslint-plugin-standard@4.0.1"
+          - "eslint-plugin-vue@7.0.0-beta.2"
+          - "eslint-plugin-cypress@2.12.1"
```
Comment (TheGreatRefrigerator) [inline .pre-commit-config.yaml:59]: are all of these needed?
Is the `@babel/eslint-parser` using all the babel things?

I think some of those are redundant anyway.  But probably we tackle them in the dependency update issue

**Review by TheGreatRefrigerator — CHANGES_REQUESTED** [review]
You could squash the last 3 chore commits to the first commit which does all the config setup.
Let's talk about the json one tomorrow

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+      - id: check-yaml
+      - id: check-json
+      - id: forbid-submodules
+      - id: double-quote-string-fixer
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: destroyed-symlinks
+      - id: detect-private-key
+      - id: pretty-format-json
+        args: ['--no-sort-keys', '--autofix', '--indent=2']
```
[the remark this answers is not in the document]
Comment (MichaelsJP) [reply .pre-commit-config.yaml:16]: JSON should always contain only unicode char sets. Adding those symbols is not a good thing:

https://docs.oracle.com/en/database/oracle/oracle-database/19/adjsn/json-character-sets-and-encoding.html#:~:text=JSON%20data%20always%20uses%20the%20Unicode%20character%20set.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (MichaelsJP) [reply .pre-commit-config.yaml:59]: Yup. Very, very hard to distinguish what's needed and what's not rn. I added all the deps bc the eslint parser was requesting them. It was an iterative process.

File: .pre-commit-config.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (MichaelsJP) [reply .pre-commit-config.yaml]: @TheGreatRefrigerator  I removed the `pretty-format-json` and replaced it with `prettier`. 
Also added a formatter for markdown and yaml. 
I think the arrays in JSON are now structured as you mentioned it above.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -5,8 +5,38 @@ on:
       - master
       - main
   pull_request:
+    branches: [ "**" ]
 jobs:
+  Run-lint:
+    strategy:
+      fail-fast: false # Good to fail all fast if one of the matrix runs fails.
```
[the remark this answers is not in the document]
Comment (MichaelsJP) [reply .github/workflows/test.yml:12]: Yep. Was still in debugging mode. Turned it on.

File: cypress/e2e/2-advanced-examples/viewport.cy.js [hunk 4]
```diff
@@ -24,33 +24,45 @@ context('Viewport', () => {
 
     // We added a cy.wait() between each viewport change so you can see
     // the change otherwise it is a little too fast to see :)
-
+    // Todo https://docs.cypress.io/guides/references/best-practices#Unnecessary-Waiting
```
[the remark this answers is not in the document]
Comment (MichaelsJP) [reply cypress/e2e/2-advanced-examples/viewport.cy.js:27]: Yup. Didn't saw that it was in the examples folder.

**Review by TheGreatRefrigerator — APPROVED** [review]

**Pull request merged**